src/app/ngrx/effects/cms.effects.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable } from '@angular/core';
+import { Router, Params, RoutesRecognized } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/switchMap';
+import { Action, Store } from '@ngrx/store';
+import { Actions, Effect } from '@ngrx/effects';
+import { EpisodeModel, EPISODE_PAGE_SIZE, EpisodeMetricsModel } from '../';
+import { selectEpisodeMetrics } from '../reducers';
+import { CmsPodcastEpisodePageAction, CmsEpisodePagePayload,
+  CmsPodcastEpisodePageSuccessAction, CmsPodcastEpisodePageFailureAction, ActionTypes } from '../actions';
+import { CmsService, HalDoc } from '../../core';
+
+@Injectable()
+export class CmsEffects {
+  episodeMetrics: EpisodeMetricsModel[] = [];
+  routeParams: Params;
+
+  @Effect()
+  loadEpisodes$: Observable<Action> = this.actions$
+    .ofType(ActionTypes.CMS_PODCAST_EPISODE_PAGE)
+    .map((action: CmsPodcastEpisodePageAction) => action.payload)
+    .switchMap((payload: CmsEpisodePagePayload) => {
+      return this.cms.follow('prx:series', {id: payload.podcast.seriesId})
+        .flatMap((series: HalDoc) => {
+          return series.followItems('prx:stories', {
+            page: payload.page,
+            per: EPISODE_PAGE_SIZE,
+            sorts: 'published_at: desc',
+            filters: 'v4',
+            zoom: 'prx:distributions'})
+          .flatMap((docs: HalDoc[]) => {
+            const chartedEpisodes = this.episodeMetrics.filter(e => e.charted).map(e => e.id);
+            // if none of the incoming episodes are already on the route, we want to add the first 5
+            const chartIncomingEpisodes = chartedEpisodes.filter(id => docs.map(doc => doc['id']).indexOf(id) !== -1).length === 0;
```

Comment:
it felt little wonky to navigate around the pages and have the only the first 5 default-selected on every new page... but I don't have any better ideas!

Reply:
Quality feedback. Noted. :+1: This is what was negotiated and landed on in the episode selection design decisions.